builder/builder-next/exporter/writer.go

```diff
@@ -0,0 +1,179 @@
+package containerimage
```

Comment:
Can we dedupe writer.go with https://github.com/moby/buildkit/blob/master/exporter/containerimage/writer.go via some util pkg?

Reply:
The reason I couldn't use it directly was that buildkit uses the `blobsum-diffid` pairs here while Moby doesn't yet have an understanding of managing layer blobs. But with a careful refactor, some parts of it can definitely be shared.